torchvision/csrc/io/decoder/stream.cpp

```diff
@@ -53,6 +56,46 @@ int Stream::openCodec(std::vector<DecoderMetadata>* metadata) {
     return ret;
   }
 
+  // multithreading heuristics
+  int max_threads = at::get_num_threads();
+  // first a safety check
+  if (num_threads > max_threads) {
+    num_threads = max_threads;
+  }
+
+  if (num_threads > 0) {
+    // if user defined, respect that
+    codecCtx_->thread_count = num_threads;
+    codecCtx_->active_thread_type = 1;
```

Comment:
Are you sure that the default behavior for `num_threads == 1` is preserved in this case? Was `active_thread_type = 1` before? Same thing for `thread_count`

Reply:
Just trying to make sure we don't have any regressions here

Reply:
It's automatically assigned to 1 in most cases, but I added a special case just in case to avoid any problems.